src/rebar3.erl

```diff
@@ -272,4 +272,14 @@ start_and_load_apps() ->
     application:start(asn1),
     application:start(public_key),
     application:start(ssl),
-    inets:start().
+    inets:start(),
+    inets:start(httpc, [{profile, hex}]),
```

Comment:
it sounds like the profile should be `rebar` given we may use httpc to also fetch things such as git or hg dependencies over https, and not just for hex.